Remotion/Validation/Core/Implementation/GenericTypeAwareValidatedTypeResolverDecorator.cs

```diff
@@ -46,9 +49,21 @@ public IValidatedTypeResolver InnerResolver
     {
       ArgumentUtility.CheckNotNull ("collectorType", collectorType);
 
-      if (TypeExtensions.CanAscribeTo (collectorType, typeof(IValidationRuleCollector<>)))
-        return TypeExtensions.GetAscribedGenericArguments (collectorType, typeof(IValidationRuleCollector<>)).Single();
-      return _validatedTypeResolver.GetValidatedType (collectorType);
+      return s_canAscribeToGenericArgumentsOfIValidationRuleCollectorCache.GetOrAdd (
```

Comment:
I'm not sure I understand why the IF and ELSE branch got merged into a single cache. It may be sensible to prevent multiple hits on CanAscribe for the ELSE branch. Unfortunately, it will also introduce a static cache for the result of an instance operation (_validatedTypeResolver.GetValidatedType).

Reply:
What would you suggest?

Reply:
I'm thinking we should do this in two parts. first, just introduce the cache but leave the code unchanged. Next, we should consider the second optimization step.

If we need more caching:
This decorator is configured to be only initialized once per IoC container so it may well be okay to use an instance cache instead of a static cache. We could also decide to use a multilayered cache, falling back to the static cache for the .net types and then caching per-instance the complete result.

Reply:
I dropped the `if` and `else` because the cache value will be created by the `valueFactory`.

Reply:
plz send meeting request :)